hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -39,31 +38,76 @@ func New(capiProviderImage string) *OpenStack {
 
 func (a OpenStack) ReconcileCAPIInfraCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster,
 	controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
-	openstackCluster := &capiopenstack.OpenStackCluster{
+	openStackCluster := &capo.OpenStackCluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      hcluster.Name,
 			Namespace: controlPlaneNamespace,
 		},
 	}
-	openstackPlatform := hcluster.Spec.Platform.OpenStack
-	if openstackPlatform != nil {
-		openstackCluster.Spec.IdentityRef = capiopenstack.OpenStackIdentityReference{
-			Name:      openstackPlatform.CloudsYamlSecret.Name,
+	openStackPlatform := hcluster.Spec.Platform.OpenStack
+	if openStackPlatform != nil {
+		openStackCluster.Spec.IdentityRef = capo.OpenStackIdentityReference{
+			Name:      openStackPlatform.CloudsYamlSecret.Name,
 			CloudName: cloudName,
 		}
 	}
 
-	if _, err := createOrUpdate(ctx, client, openstackCluster, func() error {
-		openstackCluster.Annotations = map[string]string{
-			capiv1.ManagedByAnnotation: "external",
-		}
-		// TODO(maysa): figure out why this is not being set on the object
-		openstackCluster.Status.Ready = true
+	if _, err := createOrUpdate(ctx, client, openStackCluster, func() error {
+		reconcileOpenStackCluster(hcluster, openStackCluster, openStackPlatform)
 		return nil
 	}); err != nil {
 		return nil, err
 	}
-	return openstackCluster, nil
+	return openStackCluster, nil
+}
+
+func reconcileOpenStackCluster(hcluster *hyperv1.HostedCluster, openStackCluster *capo.OpenStackCluster, openStackPlatform *hyperv1.OpenStackPlatformSpec) {
+
+	if len(openStackPlatform.Subnets) > 0 {
+		openStackCluster.Spec.Subnets = make([]capo.SubnetParam, len(openStackPlatform.Subnets))
+		for i := range openStackPlatform.Subnets {
+			openStackCluster.Spec.Subnets[i] = capo.SubnetParam{
+				ID: openStackPlatform.Subnets[i].ID,
+			}
+		}
+	} else {
+		machineNetworks := hcluster.Spec.Networking.MachineNetwork
+		openStackCluster.Spec.ManagedSubnets = make([]capo.SubnetSpec, len(machineNetworks))
+		// Only one Subnet is supported in CAPO
+		openStackCluster.Spec.ManagedSubnets[0] = capo.SubnetSpec{
```

Comment:
@EmilienM @mdbooth 
Any thoughts on the approach to handle the CAPO provisioned subnet?
Should we really consume the first CIDR from the hosted cluster and not define in the openstack platform spec? I'm enforcing the nameserver and allocationpool on that subnet.

Reply:
Lets do that for now and ask the HCP folks what they think. We definitely don't want to require the user to correctly enter the same value twice in 2 different places.